datacube_stats/mangrove.py

```diff
@@ -0,0 +1,69 @@
+import logging
+import xarray as xr
+import numpy as np
+from osgeo import ogr
+from osgeo import gdal
+from osgeo import osr
+from datacube.model import Measurement
+from .statistics import Statistic
+
+_LOG = logging.getLogger(__name__)
+
+
+class MangroveCC(Statistic):
+    def __init__(self, thresholds, shape_file, bands=None):
+        super().__init__()
+        self.thresholds = thresholds
+        if bands is None:
+            self.bands = ['extent', 'type']
+        else:
+            self.bands = bands
+        self.shape_file = shape_file
+
+    def measurements(self, input_measurements):
+        return [Measurement(name=band, dtype='int32', nodata=0, units='1') for band in self.bands]
+
+    def compute(self, data):
+        var_name = list(data.data_vars.keys())[0]
+        data[var_name].data[np.isnan(data[var_name].data)] = 0
+        rast_data = data[var_name].where(self.generate_rasterize(data[var_name]) == 1)
+
+        cover_extent = rast_data.copy(True)
+        cover_extent.data = np.zeros(cover_extent.shape)
+        cover_extent.data[rast_data.data > self.thresholds[0]] = 1
+
+        cover_type = rast_data.copy(True)
+        cover_type.data = np.zeros(cover_type.shape)
+        level_threshold = 1
+        for s_t in self.thresholds:
+            cover_type.data[rast_data.data > s_t] = level_threshold
+            level_threshold += 1
+
+        outputs = {}
+        outputs[self.bands[0]] = cover_extent
+        outputs[self.bands[1]] = cover_type
+        return xr.Dataset(outputs, attrs=dict(crs=data.crs))
+
+    def generate_rasterize(self, data):
```

Comment:
I would prefer if this is a separate function (outside `MangroveCC` class) that takes the `data` and the `shape_file`.

Up to you.

Reply:
I'll keep it here now 'cause some parameters hard-coded here really just related to the mangroves masking. Later we can figure it out if it's useful anywhere else.

Reply:
All good.